### PR TITLE
Add support for ScoreCard generation

### PR DESCRIPTION
**Pull request by shrutiparabgoogle**
In this PR:
Support for generating Scorecards

TODOs:
- `dry_run` flag across both the commands
- seeder test pattern for both commands

### CODE REVIEW

File: cmd/registry/cmd/compute/scorecard.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	defArtifact *rpc.Artifact
+	resource    patterns.ResourceInstance
+}
+
+func (task *computeScoreCardTask) String() string {
+	return "compute score " + task.resource.ResourceName().String()
+}
+
+func (task *computeScoreCardTask) Run(ctx context.Context) error {
+	return scoring.CalculateScoreCard(task.ctx, task.client, task.defArtifact, task.resource)
```
Comment (seaneganx) [inline cmd/registry/cmd/compute/scorecard.go:88]: This should use the context passed into the `Run` method.

**Review by timburks — APPROVED** [review]
LGTM - not blocking, but it would be useful to see some examples that are outside of the boilerplate of our test infrastructure and closer to the the actual user experience (YAML to YAML instead of test structs to test structs)

**Review by seaneganx — APPROVED** [review]
Approved, but please address my comment about the context.

File: cmd/registry/cmd/compute/scorecard.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+	"github.com/apigee/registry/cmd/registry/scoring"
+	"github.com/apigee/registry/connection"
+	"github.com/apigee/registry/log"
+	"github.com/apigee/registry/rpc"
+	"github.com/spf13/cobra"
+)
+
+func scoreCardCommand() *cobra.Command {
+	return &cobra.Command{
+		Use:   "scorecard (wip)",
```
[the remark this answers is not in the document]
Comment (shrutiparabgoogle) [reply cmd/registry/cmd/compute/scorecard.go]: Thanks for catching this. I took this out from `compute score` command but missed taking it out from `scorecard`.

Done.

File: cmd/registry/cmd/compute/scorecard.go (hunk 1 above)
Comment (shrutiparabgoogle) [reply cmd/registry/cmd/compute/scorecard.go]: Done.

**Comment by shrutiparabgoogle** [comment]
> LGTM - not blocking, but it would be useful to see some examples that are outside of the boilerplate of our test infrastructure and closer to the the actual user experience (YAML to YAML instead of test structs to test structs)

Yes, currently the examples that we have are through the unit and functional tests. I can create another e2e test which provide examples and also test the whole flow.

**Pull request merged**